### PR TITLE
Added functionality to the speed buttons.

### PR DESCRIPTION
**Pull request by hepfnepf**
The speed up and down buttons now work and the current speed gets displayed on the label.

### CODE REVIEW

File: Assets/UI/Scenes/TabWidgets/Buttons/MainButtons/Buttons/SpeedChangeButton.gd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+extends WidgetButton
+class_name SpeedChangeButton
+
+export (float) var speed_change = 0.0
```
Comment (pingwindev) [inline Assets/UI/Scenes/TabWidgets/Buttons/MainButtons/Buttons/SpeedChangeButton.gd]:
```suggestion
export(float) var game_speed_change = 0.0
```

File: Assets/UI/Scenes/TabWidgets/Buttons/MainButtons/Buttons/SpeedChangeButton.gd [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+func _pressed() -> void:
+	._pressed()
+
+	Global.Game.set_game_speed(Engine.time_scale+speed_change)
```
Comment (pingwindev) [inline Assets/UI/Scenes/TabWidgets/Buttons/MainButtons/Buttons/SpeedChangeButton.gd]:
```suggestion
	Global.Game.set_game_speed(Engine.time_scale + game_speed_change)
```

File: Assets/UI/Scripts/SpeedLabel.gd [hunk 3]
```diff
@@ -0,0 +1,11 @@
+extends LabelEx
+class_name SpeedLabel
```
Comment (pingwindev) [inline Assets/UI/Scripts/SpeedLabel.gd]:
```suggestion
class_name GameSpeedLabel
```

File: Assets/UI/Scripts/SpeedLabel.gd [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+func _ready() -> void:
+	call_deferred("connect_game")#waiting for Global.Game to be set
```
Comment (pingwindev) [inline Assets/UI/Scripts/SpeedLabel.gd]:
```suggestion
	call_deferred("connect_game") # waiting for Global.Game to be set
```

File: Assets/UI/Scripts/SpeedLabel.gd [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+func connect_game() -> void:
+	Global.Game.connect("speed_changed",self,"_on_speed_changed")
```
Comment (pingwindev) [inline Assets/UI/Scripts/SpeedLabel.gd]:
```suggestion
	Global.Game.connect("game_speed_changed", self, "_on_game_speed_changed")
```

File: Assets/UI/Scripts/SpeedLabel.gd [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+func _on_speed_changed(new_speed:float) -> void:
```
Comment (pingwindev) [inline Assets/UI/Scripts/SpeedLabel.gd]:
```suggestion
func _on_speed_changed(new_game_speed: float) -> void:
```

File: Assets/UI/Scripts/SpeedLabel.gd [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+	text = "%1.1fx" % new_speed
```
Comment (pingwindev) [inline Assets/UI/Scripts/SpeedLabel.gd]:
```suggestion
	text = "%1.1fx" % new_game_speed
```

File: Assets/World/Game.gd [hunk 8]
```diff
@@ -2,6 +2,9 @@ extends Spatial
 class_name Game
 
 signal notification(message_type, message_text) # int, String
+signal speed_changed(new_speed)
+
+const MAX_SPEED:float = 2.0
```
Comment (pingwindev) [inline Assets/World/Game.gd]:
```suggestion
signal game_speed_changed(new_game_speed)

const MAX_GAME_SPEED: float = 2.0
```

File: Assets/World/Game.gd [hunk 9]
```diff
@@ -79,3 +82,8 @@ func start_game() -> void:
 			pass # TODO
 
 	is_game_running = true
+
+func set_game_speed(speed:float) -> void:
+	Engine.time_scale = clamp(speed, 0.0,MAX_SPEED)
+	print_debug("Game speed changed to ", Engine.time_scale)
+	emit_signal("speed_changed",Engine.time_scale)
```
Comment (pingwindev) [inline Assets/World/Game.gd]:
```suggestion
func set_game_speed(game_speed: float) -> void:
	Engine.time_scale = clamp(game_speed, 0.0, MAX_GAME_SPEED)
	print_debug("Game speed changed to ", Engine.time_scale)
	emit_signal("game_speed_changed", Engine.time_scale)
```

**Review by pingwindev — CHANGES_REQUESTED** [review]
Please fix the marked code style issues.
I suggest to rename all instances of `Speed`/`speed`/`SPEED` into `GameSpeed`/`game_speed`/`GAME_SPEED` to avoid ambiguity. I've marked all occurrences that I could locate. Also the script names and the nodes from the TabWidget should be updated to clearly reflect their use.

**Comment by LinuxDonald** [comment]
@hepfnepf will you do the needed changes please?

**Review by pingwindev — APPROVED** [review]

**Review by pingwindev — APPROVED** [review]

**Pull request merged**